doc/Clock commands/clock_managment_hld.md

```diff
@@ -0,0 +1,180 @@
+	* 1.1. [Revision](#Revision)
+	* 1.2. [Scope](#Scope)
+	* 1.3. [Definitions/Abbreviations](#DefinitionsAbbreviations)
+	* 1.4. [Overview](#Overview)
+	* 1.5. [Requirements](#Requirements)
+		* 1.5.1. [Functional requirements](#FunctionalRequirements)
+		* 1.5.2. [Configuration and Management Requirements](#ConfigurationManagementRequirements) 
+* 2. [Design](#Design)
+	* 2.1. [High-Level Design](#High-LevelDesign)
+* 3. [Configuration and management](#Configurationandmanagement)
+	* 3.1. [ConfigDB Tables](#ConfigDBTables)
+	* 3.2. [CLI/YANG model](#CLIYANGmodel)
+		* 3.2.1. [Yang model](#Yangmodel)
+		* 3.2.2. [CLI](#Climodel)
+* 4. [Test Plan](#TestPlan)
+		* 4.1. [Unit Test cases](#UnitTestcases)
+
+
+### 1.1 <a name='Revision'></a>Revision
+
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 01/2023 | Meir Renford	 | Phase 1 Design     |
+
+###  1.2. <a name='Scope'></a>Scope
+
+This document will address the high level design for NVOS clock commands:
+1.	Set/show date-time command
+2.	Set/show timezone command
+
+
+### 1.3 <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations 
+
+N/A
+
+### 1.4 Overview 
+
+The clock commands allow to set and review the current time parameters of the system - including: time, date and timezone. 
+
+### 1.5 Requirements
+####  1.5.1. <a name='FunctionalRequirements'></a>Functional requirements
+
+1. Any time configuration that will change in the system will change the system time. 
+
+####  1.5.2. <a name='ConfigurationManagementRequirements'></a>Configuration and Management Requirements
+The requirements from the module are: 
+1.	Set and show the system time and date.
+2.	Set and show the system timezone.
+
+
+##  2 <a name='Design'></a>Design
+
+###  2.1 <a name='High-LevelDesign'></a>High-Level Design
+
+The design of this feature is based on Linux command of <b>timedatectl</b>.<BR>
+(man page for timedatectl: https://man7.org/linux/man-pages/man1/timedatectl.1.html)
+
+All set operations are based on this command, executing directly in linux upon user CLI execution.
+The time/date/timezone are configured directly to Linux, and is persistent upon any reboot.
+
+System state of time/date/timezone will appear in 1 line as an output of existing "show clock" command.
+
+Set operations will be divided into 2 different commands:
+
+1. config clock set-timezone <timezone> (to set timezone command)
+	* will get single input as a string, and validate it is valid as part of ("timedatectl list-timezones" command).
+	Validation will be done either by YANG model, or (if not possible) by issueing relevant error to log.
+	* Value will be stored in db for future upgrade operations.
+	  Value is persistent upcon reboot.
+	* Linux timedatectl with set-timezone flag will be called.
+	  e.g. timedatectl set-timezone "Asia/Kolkata"
+	* <b>In case NTP is enabled -> timezone configuration is allowed and overrides the current time.</b>
```

Comment:
Can this data & timezone change apply will take care of all the services/dockers date & timezone setting? ex: FRR docker

Reply:
NTP is based on UTC only, and timezone provides the offset to generate the clock time. So even without NTP enabled, timezone configuration is allowed and will override the current time if from a different offset?

Reply:
Yes, it will take care of all the services/dockers.

Reply:
Yes. NTP will override the current time if from a different offset.

Reply:
> Yes. NTP will override the current time if from a different offset.

I'm still not clear about what is the "current time" here, and how NTP overrides it if a different timezone is configured. For example, originally timezone is UTC.

root@sonic:~# date
Thu 27 Apr 2023 12:11:29 AM UTC

root@sonic:~# timedatectl
               Local time: Thu 2023-04-27 00:12:02 UTC
           Universal time: Thu 2023-04-27 00:12:02 UTC
                 RTC time: Thu 2023-04-27 00:12:02
                Time zone: Etc/UTC (UTC, +0000)
...

root@sonic:~# show ntp
MGMT_VRF_CONFIG is not present.
     remote           refid      st t when poll reach   delay   offset  jitter
==============================================================================
*10.199.39.10    10.254.225.252   2 u   42   64  377    0.131    4.219   0.565


Then change to new timezone (US/Pacific) 
//////////////////////////////////////////////////
root@sonic:~# date
Wed 26 Apr 2023 05:16:49 PM PDT

root@sonic:~# timedatectl
               Local time: Wed 2023-04-26 17:16:55 PDT
           Universal time: Thu 2023-04-27 00:16:55 UTC
                 RTC time: Thu 2023-04-27 00:16:55
                Time zone: US/Pacific (PDT, -0700)
...

root@sonic:~# show ntp
MGMT_VRF_CONFIG is not present.
     remote           refid      st t when poll reach   delay   offset  jitter
==============================================================================
*10.199.39.10    10.254.225.252   2 u    2   64  377    0.132    4.981   1.026

 Do you mean the new  timezone adds an offset to the original synchronized time?